src/pkgmt/versioneer.py

```diff
@@ -128,9 +141,10 @@ def version(
             "Commit them or discard them and try again.\nDetected files:"
             f"\n{pending.decode()}"
         )
-
-    if version_package:
-        versioner = VersionerNonSetup(version_package, project_root=project_root)
+    cfg = load()
+    version_file = cfg.get("version_file", None)
+    if version_file:
+        versioner = VersionerNonSetup(version_file, project_root=project_root)
```

Comment:
seems like the difference between VersionerSetup and VersionerNonSetup is that VersionerSetup will automatically lookup the `__init.__py` while VersionerNonSetup won't. 

maybe it's best to unify the logic, say, in a Versioner class? we can extract the logic in VersionerSetup to lookup the `__init__.py`

something like this:

1. if version_file exists, then use it
2. if version_file does not exist, call a function to lookup `__init__.py`
3. then initialize Versioner

Reply:
So should I remove the `AbstractVersioner`? and replace it with `Versioner`

Reply:
yes, we won't need it anymore